ads7138.c

```diff
@@ -220,8 +222,16 @@ static int ads7138_probe(struct i2c_client *client,
 		goto out;
 	}
 
-	/* Get system status */
-	status = ads7138_read_reg(data, ADS7138_REG_SYSTEM_STATUS);
+	for (status_counter = 0; status_counter < 6; ++status_counter) {
```

Comment:
Thanks for your contribution! Apparently, a 5 ms delay might be needed after power on as well (section 7.9 of the datasheet). Also it would be better to use the time_before function instead of a loop with a counter. I would suggest creating a function like this one:
```
static int ads7138_wait_ready(const struct ads7138_data *data)
{
    unsigned long timeout = jiffies + msecs_to_jiffies(5);
    int status;

    do {
        status = ads7138_read_reg(data, ADS7138_REG_SYSTEM_STATUS);
        if (status >= 0)
            break;

        usleep_range(50, 100);

    } while (time_before(jiffies, timeout));

    dev_dbg(dev, "ads7138 status 0x%x\n", status);
    return status;
}
```

And then just call it before and after the reset:
```
/* Wait for power-up */
status = ads7138_wait_ready(data);
if (status < 0) {
    dev_err(dev, "Failed to read system status after power-up\n");
    goto out;
}

/* Software reset */
ret = ads7138_write_reg(data, ADS7138_REG_GENERAL_CFG, ADS7138_GENERAL_CFG_RST);
if (ret < 0) {
    dev_err(dev, "Failed to reset\n");
    goto out;
}

/* Wait for reset to complete */
status = ads7138_wait_ready(data);
if (status < 0) {
    dev_err(dev, "Failed to read system status after reset\n");
    goto out;
}
```

Reply:
I will make the changes you suggested and test them.

Reply:
Changed and tested. Added dev parameter to wait_ready so dev_err would work.